### DISCUSSION TITLE
Unable to update Back Order text with blocks but can with classic cart.

### DISCUSSION THREAD

**Discussion by emcgal** [woocommerce/woocommerce#45031] [WooCommerce, Blocks, and Site Editing]
Hi @alexflorisca, it is me again :-)

I am not sure if this is another potential issue with blocks, please just close off if it isn't and I will wait for the forum. 

I finally got a chance to try and get a stab at changing the text on backorders in the basket this evening. I have logged it on the plugin forum and with Woocommerce Support but I was thinking, it might be an issue as I have it working with the classic cart so best to share it. 
Plugin forum
https://wordpress.org/support/topic/trying-to-change-the-text-on-back-orders-in-the-cart-with-blocks/

## Describe the bug
I am trying to change the text in the checkout process for Back orders to Pre-order which makes more sense for customer (Back order is very much a retail term) and I have it on single product pages and this code is working on the classic [woocommerce_cart] but not on blocks.

## To reproduce

1. Simply add a back order product to the basket.
2. Please note I do have both the classic and block cart on the basket page trying to figure this out.
3. A back order product to test
https://emerc38.sg-host.com/product/clifdens-churches-wall-panel/

## Expected behavior
I want to change all the back order text to state PRE-ORDER which is more user friendly plus consistency from a UX perspective.

## Screenshots
Fine here using FiboSearch plugin
https://prnt.sc/2090zx4

Fine with the classic Woocommerce cart
https://prnt.sc/208zims

But I have blocks on the same page and it isn't updating. Still showing default text.
https://prnt.sc/208znwc

And not working in the blocks checkout either.
https://prnt.sc/2090krt

### WordPress (please complete the following information):
* Core version: 5.8.2
* WooCommerce version: Version 5.9.0
* WooCommerce Blocks version: Version 6.3.2
* Site language: EN
* Any other plugins installed: several but I don't think applicable to this.

### Desktop (please complete the following information):
* OS: , Windows 10
* Browser: Chrome Version 96.0.4664.45

## Additional context
My code for this:

function woocommerce_custom_cart_item_name( $_product_title, $cart_item, $cart_item_key ) {
	if ( $cart_item['data']->backorders_require_notification() && $cart_item['data']->is_on_backorder( $cart_item['quantity'] ) ) {
     $_product_title .=  __( '<br /><strong>PRE-ORDER ITEM</strong>', 'woocommerce' ) ;
   }
   return $_product_title;
}
add_filter( 'woocommerce_cart_item_name', 'woocommerce_custom_cart_item_name', 10, 3);

**Comment by senadir** [comment]
Hey, this is not yet an option we provide, however, you can filter the translation string with this JS code, this is available as of WordPress 5.7.0.

```js
function myPluginGettextFilter( translation, text, domain ) {
	if ( text === 'Available on backorder' ) {
		return 'Pre-order item';
	}
	return translation;
}

// Adding the filter
wp.hooks.addFilter(
	'i18n.gettext_woo-gutenberg-products-block',
	'my-plugin/override-backorder-text',
	myPluginGettextFilter
);
```

**Comment by emcgal** [comment]
Thanks a mil @senadir

Getting an error at the moment (syntax error, unexpected ',', expecting variable (T_VARIABLE)) but will try and get it going.

**Comment by senadir** [comment]
Depends on how you're printing the code maybe, I tested the code above and it worked for me.

**Comment by emcgal** [comment]
@senadir I just added it to the functions.php, commented out the other and added the domain. 
Was I supposed to change anything else?

**Comment by senadir** [comment]
This code worked fine for me

```php
add_action( 'wp_enqueue_scripts', function () {
	wp_add_inline_script(
		'wc-cart-block-frontend',
		"
		function myPluginGettextFilter( translation, text, domain ) {
			if ( text === 'Available on backorder' ) {
				return 'Pre-order item';
			}
			return translation;
		}

		// Adding the filter
		wp.hooks.addFilter(
			'i18n.gettext_woo-gutenberg-products-block',
			'my-plugin/override-backorder-text',
			myPluginGettextFilter
		);
		",
		'after'
	);
});
```

**Comment by senadir** [comment]
No changes needed except on the text level. The domain `woo-gutenberg-products-block` is correct.

**Comment by emcgal** [comment]
@senadir Super that is working perfectly ..... in the basket, thanks so much but unfortunately not in the checkout. Still marked as back order there :-(
ie: https://prnt.sc/2090krt

Also, it is OK to use with the code I have written to show in the basket notification?
ie this one https://prnt.sc/2090zx4

**Comment by senadir** [comment]
> Super that is working perfectly ..... in the basket, thanks so much but unfortunately not in the checkout. Still marked as back order there :-(

Yes, that's because it's only included with `wc-cart-block-frontend`. Duplicate the code and use `wc-checkout-block-frontend`.

> Also, it is OK to use with the code I have written to show in the basket notification?

Yes, they should work both fine. Once you switch to Mini Cart in the future, make sure to do the same as well.

**Comment by emcgal** [comment]
Sorry for the delay, it was one of those days. 

Thanks so much! I have got so rusty with code at this stage an never really got familiar with blocks.

Really appreciate your help.

Thanks again @senadir
